### PR TITLE
merge accelerators

### PR DESCRIPTION
**Pull request by abdulajet**
## Pull request checklist

- [x] All tests pass. Demo project builds and runs.
- [x] I have resolved any merge conflicts. 

TODO: Update README
TODO: Update travis.yml 
See https://github.com/opentok/accelerator-core-ios/pull/66

## What's in this pull request?

This PR brings in the TextChat and Annotation Accelerators into the Core project. It is quite a big PR but you can review commit by commit. The first two commits are mainly copying in the text chat and annotation projects. Below is a screenshot of the project structure. The sample projects and test targets have been copied over with the source files for the accelerators under the core folder.

image

### CODE REVIEW

File: OTAcceleratorCore.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -543,23 +1378,111 @@
...
+			);
+			name = "[CP] Check Pods Manifest.lock";
+			outputFileListPaths = (
+			);
+			outputPaths = (
+				"$(DERIVED_FILE_DIR)/Pods-OTAnnotationSample-checkManifestLockResult.txt",
+			);
+			runOnlyForDeploymentPostprocessing = 0;
+			shellPath = /bin/sh;
+			shellScript = "diff \"${PODS_PODFILE_DIR_PATH}/Podfile.lock\" \"${PODS_ROOT}/Manifest.lock\" > /dev/null\nif [ $? != 0 ] ; then\n    # print error to STDERR\n    echo \"error: The sandbox is not in sync with the Podfile.lock. Run 'pod install' or update your CocoaPods installation.\" >&2\n    exit 1\nfi\n# This output is used by Xcode 'outputs' to avoid re-running this script phase.\necho \"SUCCESS\" > \"${SCRIPT_OUTPUT_FILE_0}\"\n";
```
Comment (rperezcubero) [inline OTAcceleratorCore.xcodeproj/project.pbxproj:1407]: I think it is a good idea to call `pod deintegrate` before commiting the changes so the code in the repo is "free" from cocoapods modification

File: OTAcceleratorCore/OTAnnotation/AnnLoggingWrapper.m [hunk 2]
```diff
@@ -0,0 +1,32 @@
+//
+//  AnnLoggingWrapper.m
+//  OTAnnotationAccPackKit
+//
+//  Created by Xi Huang on 8/25/16.
```
Comment (rperezcubero) [inline OTAcceleratorCore/OTAnnotation/AnnLoggingWrapper.m:5]: I would say we can remove this headers

File: OTAcceleratorCore/OTAnnotation/Category/JSON.h [hunk 3]
```diff
@@ -0,0 +1,15 @@
+//
+//  JSON.h
+//
+//  Copyright © 2016 Tokbox, Inc. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
```
Comment (rperezcubero) [inline OTAcceleratorCore/OTAnnotation/Category/JSON.h:7]: Do we need a class like this?
It looks like JSON parsing is already in the iOS SDK.

**Comment by rperezcubero** [comment]
Started to do some reviews, but relized that this PR was not about reviewing the code of the _old_ accelerator packs we had so I removed those comments.

Just left one about the `pop deintegrate` but in any case feel free to merge as it is if you decide to.

**Review by rperezcubero — APPROVED** [review]

**Comment by abdulajet** [comment]
Thanks, ill run the `pod deintegrate`

**Review by marinaserranomontes — APPROVED** [review]
LGTM

**Review by rperezcubero — APPROVED** [review]

**Pull request merged**